src/Assets/Asset.php

```diff
@@ -1214,6 +1339,38 @@ public function register() {
 		Assets::init()->register_in_wp( $this );
 	}
 
+	/**
+	 * Register the asset along with registering a CSS asset from the same directory.
+	 *
+	 * @since 1.3.1
+	 *
+	 * @param string|callable ...$dependencies The dependencies to add to the cloned asset.
+	 *
+	 * @return void
+	 */
+	public function register_with_css( ...$dependencies ) {
+		$this->prefix_asset_directory( false );
+		$this->register();
+		$asset = $this->clone_to( 'css', ...$dependencies );
+		$asset->register();
+	}
+
+	/**
+	 * Register the asset along with registering a JS asset from the same directory.
+	 *
+	 * @since 1.3.1
+	 *
+	 * @param string|callable ...$dependencies The dependencies to add to the cloned asset.
+	 *
+	 * @return void
+	 */
+	public function register_with_js( ...$dependencies ) {
+		$this->prefix_asset_directory( false );
+		$this->register();
+		$asset = $this->clone_to( 'js', ...$dependencies );
+		$asset->register();
+	}
```

Comment:
Why not return the clone? Feels like it gives the system some extra flexibility.

Reply:
Oh, actually it'd probably be better to return the original asset